R/jhu_data.R

```diff
@@ -72,74 +99,123 @@ load_jhu_data <- function(
+          county_results <- county_results %>%
+            dplyr::filter(location %in% selected_counties)
+        }
+        
+        results <- dplyr::bind_rows(results, county_results)
+      }
+      
+    # summarized results for state level
+    if ("state" %in% spatial_resolution | length(selected_states) > 0 ) {
+      if (length(selected_states) > 0) {
+          # pull state names from fips code
+          states_to_keep <- valid_locations %>%
+            dplyr::filter(location %in% selected_states) %>%
+            dplyr::pull(location_name)
+      } else {
+        states_to_keep <- c(
+          "Alabama", "Alaska", "American Samoa", "Arizona", "Arkansas",
+          "California", "Colorado", "Connecticut", "Delaware",
+          "District of Columbia", "Florida", "Georgia", "Guam", "Hawaii", "Idaho",
+          "Illinois", "Indiana", "Iowa", "Kansas", "Kentucky", "Louisiana", "Maine",
+          "Maryland", "Massachusetts", "Michigan", "Minnesota",
+          "Mississippi", "Missouri", "Montana", "Nebraska", "Nevada",
+          "New Hampshire", "New Jersey", "New Mexico", "New York",
+          "North Carolina", "North Dakota", "Northern Mariana Islands",
+          "Ohio", "Oklahoma", "Oregon", "Pennsylvania", "Puerto Rico",
+          "Rhode Island", "South Carolina", "South Dakota", "Tennessee",
+          "Texas", "Utah", "Vermont", "Virgin Islands", "Virginia",
+          "Washington", "West Virginia", "Wisconsin", "Wyoming")
+      }
+        
+      state_results <- jhu_data %>%
+        dplyr::filter(Province_State %in% states_to_keep) %>%
+        dplyr::mutate(location_name = Province_State) %>%
+        dplyr::group_by(location_name, date) %>%
+        dplyr::summarize(cum = sum(cum)) %>%
+        dplyr::group_by(location_name) %>%
+        dplyr::mutate(inc = diff(c(0, cum))) %>%
+        dplyr::ungroup() %>%
+        dplyr::left_join(
+          valid_locations %>% dplyr::filter(nchar(location) == 2),
+          by = "location_name") %>%
+        dplyr::select(location, date, cum, inc)
+        
+      results <- dplyr::bind_rows(results, state_results)
+    }
+      
+    # summarized results for national level
+    if ("national" %in% spatial_resolution | length(selected_us) == 1) {
+      # because we don't filter on states_to_keep as above, we are off by a total
+      # of 3 deaths attributed to Diamond Princess.
+      national_results <- jhu_data %>%
+        dplyr::group_by(date) %>%
+        dplyr::summarize(cum = sum(cum)) %>%
+        dplyr::ungroup() %>%
+        dplyr::mutate(
+          inc = diff(c(0, cum)),
+          location = "US") %>%
+        dplyr::select(location, date, cum, inc)
+        
+      results <- dplyr::bind_rows(results, national_results)
+    }
+  } else if (hub[1] == "ECDC"){
+      results <- jhu_data %>%
+        dplyr::rename(location = `Country/Region`) %>%
+        dplyr::group_by(location) %>%
+        dplyr::mutate(inc = diff(c(0, cum))) %>%
```

Comment:
need to do some stuff here to handle countries with multiple rows that should be aggregated